### PR TITLE
rename estimators module to regression 

### PR DESCRIPTION
**Pull request by aagoumbala**
# Description

- Renaming estimators module to regression. Update documentation with new module name
- Modification of the documentation architecture : change section title "DOCUMENTATION" to "REGRESSION".Move the API to the last subsection renaming it to "API AND EXAMPLES"

Fixes #(issue)

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `flake8 . --exclude=doc`
- [x] Typing passes successfully : `mypy mapie examples --strict`
- [x] Unit tests pass successfully : `pytest -vs --doctest-modules mapie`
- [x] Coverage is 100% : `pytest -vs --doctest-modules --cov-branch --cov=mapie --pyargs mapie`
- [x] Documentation builds successfully : `cd doc; make clean; make html`

### CODE REVIEW

**Comment by vtaquet** [comment]
Très bonne première PR Akim :-) . Quelques questions:

Pourrais-tu t'ajouter dans la liste des contributeurs dans AUTHORS.rst ?
Pourrais-tu initier la prochaine release (0.2.4 avec date XX pour le moment) en ajoutant une première ligne du style "Rename estimators.py to regression.py ?
Afin de préparer la séparation de la doc entre regression et classification, pourrais-tu transformer le titre de section "DOCUMENTATION" en "REGRESSION" ? Je déplacerais également l'API dans la dernière sous-section en la renommant en "API AND EXAMPLES"

**Comment by aagoumbala** [comment]
J'ai ajouté les modifications suggéré sur ce [commentaire](https://github.com/scikit-learn-contrib/MAPIE/pull/78#issuecomment-886487237)

**Review by gmartinonQM — APPROVED** [review]

File: HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,12 @@
 History
 =======
 
+0.2.4 (XXXX-XX-XX)
+------------------
+
+* Renaming estimators module to Regression
```
Comment (vtaquet) [inline HISTORY.rst:8]: Changer en "Renaming `estimators.py` module to `regression.py`

Comment (aagoumbala) [reply HISTORY.rst]: c'est changé

**Pull request merged**